main.c

```diff
@@ -191,12 +191,19 @@ static int init(void *data) {
   gdk_window_set_cursor(gtk_widget_get_window(top_window),cursor_watch);
 
   //
-  // Let WDSP (via FFTW) check for wisdom file in current dir
+  // Let WDSP (via FFTW) check for wisdom file in home dir
   // If there is one, the "wisdom thread" takes no time
   // Depending on the WDSP version, the file is wdspWisdom or wdspWisdom00.
   // sem_trywait() is not elegant, replaced this with wisdom_running variable.
   //
-  char *c=getcwd(wisdom_directory, sizeof(wisdom_directory));
+  char *c=getenv("HOME");
+  if (c) {
+      strncpy(wisdom_directory, c, 1023);
```

Comment:
I would use sizeof(wisdom_directory)-1 for size.

Reply:
Wouldn't be something like ``~/.local/state/pihpsdr`` directory more appropriate? I am not sure what exactly that wisdom file is for, but storing it directly in $HOME does not seem correct in any case. At least ``~/.pihpsdr`` would be expected by me.

Reply:
sizeof() is better, I will update it.

Subdirectory would require directory creation and error handling, i.e. more code. Moreover, IMHO the wisdom file can be shared across any application using WDSP library. Putting it into HOME prevents the application from recalculation of it when starting from different directories. Recalculation can take a lot of time. I don't know whether it can be calculated on the builder (there could be utilized target CPU optimizations, but I am not sure about it) and put e.g. to /var, but these are bigger changes that should be handled by WDSP upstream.